File: frontend/src/api/terminal.ts
export type TerminalResult = {
  command: string;
  cwd: string;
  previous_cwd: string;
  exit_code: number;
  stdout: string;
  stderr: string;
};

const apiBaseUrl = import.meta.env.VITE_API_BASE_URL ?? "http://localhost:8000";

export async function executeTerminalCommand(command: string): Promise<TerminalResult> {
  const response = await fetch(`${apiBaseUrl}/api/terminal/execute`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ command, timeout_seconds: 120 }),
  });
  if (!response.ok) {
    const error = await response.json().catch(() => null);
    throw new Error(error?.detail ?? "No se pudo ejecutar el comando");
  }
  return response.json();
}

File: backend/app/schemas/terminal.py
from pydantic import BaseModel, Field


class TerminalCommand(BaseModel):
    command: str = Field(min_length=1)
    timeout_seconds: int = Field(default=120, ge=1, le=1800)


class TerminalResult(BaseModel):
    command: str
    cwd: str
    previous_cwd: str
    exit_code: int
    stdout: str
    stderr: str
